src/analytics/application/missing-indexables-collector.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace Yoast\WP\SEO\Analytics\Application;
+
+use Yoast\WP\SEO\Analytics\Domain\Missing_Indexable_Bucket;
+use Yoast\WP\SEO\Analytics\Domain\Missing_Indexable_Count;
+use Yoast\WP\SEO\Actions\Indexing\Indexation_Action_Interface;
+
+/**
+ * Manages the collection of the missing indexable data.
+ */
+class Missing_Indexables_Collector implements \WPSEO_Collection {
+
+	/**
+	 * All the indexation actions.
+	 *
+	 * @var array<Indexation_Action_Interface> $indexation_actions
+	 */
+	private $indexation_actions;
+
+	/**
+	 * The collector constructor.
+	 *
+	 * @param Indexation_Action_Interface ...$indexation_actions All the Indexation
+	 *                                                                                          actions.
+	 */
+	public function __construct( Indexation_Action_Interface ...$indexation_actions ) {
+		$this->indexation_actions = $indexation_actions;
+		$this->add_additional_indexing_actions();
```

Comment:
I had this before, but while talking with Diede we decided it would be nicer to have it this way so the flow is not changed just to accompany Premium. In this scenario premium and or add-ons can provide indexing actions and because they all have the same interface they can just follow the normal flow. This feels more SOLID to me since now I have no extra code in my business logic to `also` include premium. I prepare my actions in the constructor and the rest of the logic can just assume everything is present. If we would add other methods later on in this collector we can still assume we have all the information present :)

Let me know if this makes sense